internal/cli/create_repo.go

```diff
@@ -0,0 +1,85 @@
+package cli
+
+import (
+	"context"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/tilt-dev/tilt/internal/analytics"
+	engineanalytics "github.com/tilt-dev/tilt/internal/engine/analytics"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+// A human-friendly CLI for creating extension repos.
+type createRepoCmd struct {
+	helper *createHelper
+
+	ref string
+}
+
+var _ tiltCmd = &createRepoCmd{}
+
+func newCreateRepoCmd() *createRepoCmd {
+	helper := newCreateHelper()
+	return &createRepoCmd{
+		helper: helper,
+	}
+}
+
+func (c *createRepoCmd) name() model.TiltSubcommand { return "create" }
+
+func (c *createRepoCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "repo NAME URL [ARG...]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Register an extension repository.",
+		Long: `Register a repository for loading Tilt extensions.
+
+Tilt supports both git-hosted and local filesystem repositories.
+`,
+		Args: cobra.MinimumNArgs(2),
+		Example: `
+tilt create repo default https://github.com/tilt-dev/tilt-extensions
+tilt create repo default file:///home/user/src/tilt-extensions
+tilt create repo default https://github.com/tilt-dev/tilt-extensions --ref=SHA
+`,
+	}
+
+	cmd.Flags().StringVar(&c.ref, "ref", "",
+		"Git reference to sync the repository to.")
+
+	c.helper.addFlags(cmd)
+
+	return cmd
+}
+
+func (c *createRepoCmd) run(ctx context.Context, args []string) error {
+	a := analytics.Get(ctx)
+	cmdTags := engineanalytics.CmdTags(map[string]string{})
+	a.Incr("cmd.create-repo", cmdTags.AsMap())
```

Comment:
Might be helpful to have a `url_type` tag with whether it's a `file`/git URL. (Only other possible data point that could be added would be a boolean indicating whether a ref was pinned, but I dunno if there's any value in that?)

Reply:
i'm going to try to add some analytics deeper in the extension reconciler that i think will address exactly this